extension/httpfs/httpfs-extension.cpp

```diff
@@ -10,21 +11,37 @@ namespace duckdb {
 static void LoadInternal(DatabaseInstance &instance) {
 	S3FileSystem::Verify(); // run some tests to see if all the hashes work out
 	auto &fs = instance.GetFileSystem();
+
 	fs.RegisterSubSystem(make_unique<HTTPFileSystem>());
-	fs.RegisterSubSystem(make_unique<S3FileSystem>());
+	fs.RegisterSubSystem(make_unique<S3FileSystem>(*instance.GetStorageManager().buffer_manager));
```

Comment:
`BufferManager::GetBufferManager(*instance)`

Then the `storage_manager.hpp` include can also be removed.